pkg/controller/operators/olm/apiservices.go

```diff
@@ -221,6 +221,44 @@ func (a *Operator) areAPIServicesAvailable(descs []v1alpha1.APIServiceDescriptio
 	return true, nil
 }
 
+func (a *Operator) areAPIServicesNotPresentOrAdoptable(csv *v1alpha1.ClusterServiceVersion) bool {
```

Comment:
This makes sense, but I wonder if we can avoid duplicating the work. What do you think about doing something like this: https://gist.github.com/njhale/f63eecc64a8a78197b0c43d634667aeb ?